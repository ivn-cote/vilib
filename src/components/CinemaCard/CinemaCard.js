import React from 'react';
import { PropTypes } from 'prop-types';
import {
  Card,
  CardActions,
  CardMedia,
  CardTitle,
  CardText
} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import styles from './CinemaCard.scss';

const CinemaCard = ({ className, title, quote, producer, year, description }) => (
  <Card className={`${styles.CinemaCard} ${className}`}>
    <CardMedia
      overlay={<CardTitle title={quote} />}
    >
      <img className={styles.CinemaCardImage} src={require('./assets/film-1.svg')} />
    </CardMedia>
    <CardTitle title={title} subtitle={`${producer} ${year}`} />
    <CardText>
      {description}
    </CardText>
    <CardActions>
      <FlatButton label="Update" />
      <FlatButton label="Delete" />
    </CardActions>
  </Card>
);

CinemaCard.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  producer: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

CinemaCard.defaultProps = {
  className: ''
};
export default CinemaCard;
