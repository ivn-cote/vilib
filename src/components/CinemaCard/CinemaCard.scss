.CinemaCard {

}

.CinemaCardImage {
  height: 200px;
}
