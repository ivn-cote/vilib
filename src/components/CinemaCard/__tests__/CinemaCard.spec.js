import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import CinemaCard from '../index';

describe('<CinemaCard />', () => {
  it('should have a proper markup', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <CinemaCard
          title="title"
          quote="quote"
          producer="producer"
          year="year"
          description="description"
        />
      </MuiThemeProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
