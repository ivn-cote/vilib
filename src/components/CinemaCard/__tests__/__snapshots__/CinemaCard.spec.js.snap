// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CinemaCard /> should have a proper markup 1`] = `
<div
  className="CinemaCard "
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      style={
        Object {
          "muiPrepared": true,
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <img
          className="CinemaCardImage"
          src="test-file-stub"
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "muiPrepared": true,
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "background": "rgba(0, 0, 0, 0.54)",
                "bottom": 0,
                "left": 0,
                "muiPrepared": true,
                "paddingTop": 8,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "padding": 16,
                  "position": "relative",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "display": "block",
                    "fontSize": 24,
                    "lineHeight": "36px",
                    "muiPrepared": true,
                  }
                }
              >
                quote
              </span>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.54)",
                    "display": "block",
                    "fontSize": 14,
                    "muiPrepared": true,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "block",
            "fontSize": 24,
            "lineHeight": "36px",
            "muiPrepared": true,
          }
        }
      >
        title
      </span>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "muiPrepared": true,
          }
        }
      >
        producer year
      </span>
    </div>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    >
      description
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 8,
          "position": "relative",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Update
          </span>
        </div>
      </button>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Delete
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
